Tests/LocalLLMExecutionDemo/LocalLLMExecutionDemo/Onboarding/LocalLLMDownloadManager.swift

```diff
@@ -0,0 +1,85 @@
+//
+// This source file is part of the SpeziML open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+
+/// Manages the download of the LLM to the local device.
+final class LocalLLMDownloadManager: NSObject, ObservableObject, Sendable, URLSessionDownloadDelegate {
```

Comment:
I think it would be very beneficial if we can move this code and related functionality (and maybe even UI) into a separate target that builds on Spezi Local LLM 👍